openapi/index.yaml

```diff
@@ -15,117 +15,35 @@ schemes:
   - https
 paths:
   /online-merchants:
-    parameters:
-      - name: merchantName
-        in: query
-        description: Text used for searching merchants by their name
-        required: false
-        type: string
-      - name: productCategories
-        in: query
-        description: List of product categories to filter the results
-        required: false
-        collectionFormat: csv
-        type: array
-        items:
-          type: string
-          enum:
-            - entertainments
-            - travels
-            - transportation
-            - connectivity
-            - books
-            - arts
-            - sports
-            - health
-      - name: page
-        in: query
-        description: page of result
-        required: false
-        type: integer
-        default: 0
-      - name: pageSize
-        in: query
-        description: elements per page
-        required: false
-        type: integer
-        default: 100
-    get:
+    post:
       operationId: getOnlineMerchants
       summary: List of online merchants
       description: |
         List of online merchants alphabetically ordered
+      parameters:
+        - in: body
+          name: searchParameter
```

Comment:
we can safely remove name so the entire body maps over the entire request

Reply:
Removed